Sources/SystemMetrics/Docs.docc/Proposals/SSM-0001.md

```diff
@@ -0,0 +1,145 @@
+# SSM-0001: `swift-metrics-extras` revised public API to support 1.0 roadmap
+
+As part of the 1.0 release we would like to revise the public API to simplify adoption and maintenance:
+* Reduce public API to a minimum to make it easier to support more platforms and metrics gathering strategies.
+* Decouple `swift-metrics-extras` public interface from the global `MetricsSystem`.
+* Add compatibility with `swift-service-lifecycle`
+
+## Overview
+
+- Proposal: SSM-0001
+- Author(s): [Vladimir Kukushkin](https://github.com/kukushechkin)
+- Status: **Awaiting Review**
+- Issue: [apple/swift-metrics-extras#64](https://github.com/apple/swift-metrics-extras/issues/64)
+- Implementation:
+    - [apple/swift-metrics-extras#66](https://github.com/apple/swift-metrics-extras/pull/66)
+- Related links:
+    - [Swift Evolution](https://www.swift.org/swift-evolution/)
+
+### Introduction
+
+This proposal aims to change the public interface of the package to bring it in line with the latest best practices in the Swift ecosystem and reduce the API to a minimum focused on automatic background gathering of system metrics for the supported platforms.
+
+### Motivation
+
+`swift-metrics-extras` provides only one component — `SystemMetrics`, which periodically collects process system metrics in the background. The monitoring is a separate component, which depends on the `MetricsSystem`, but is not part of it. While it is possible to use `SystemMetrics` as an independent object, the user would need to maintain its lifecycle manually, and it still requires the globally initialized `MetricsSystem`.
+Coupling with the `MetricsSystem` bootstrapping is unnecessary and prevents users from using different backends for different metrics, for example, if `SystemMetrics` should be sent to a different backend. This coupling also makes it hard to unit test the package.
+
+### Proposed solution
+
+To solve the unnecessary coupling with the `MetricsSystem`, I propose to change the interface of the `SystemMetrics` — a new `SystemMetricsMonitor` object compatible with the `swift-service-lifecycle` `Service` protocol to run the periodic metrics collection and reporting in the background for the whole lifecycle of the parent service.
+
+This is how it can be integrated into a service:
+
+```
+import SystemMetrics
+import ServiceLifecycle
+import UnixSignals
+import Metrics
+
+@main
+struct Application {
+    static let logger = Logger(label: "Application")
+    static let metrics = MyMetricsBackendImplementation()
+
+    static func main() async throws {
+        MetricsSystem.bootstrap(metrics)
+
+        let service = FooService()
+        let systemMetricsMonitor = SystemMetricsMonitor()
+        let serviceGroup = ServiceGroup(
+            services: [service, systemMetricsMonitor],
+            gracefulShutdownSignals: [.sigterm],
```

Comment:
Please make the signals match https://github.com/swiftlang/swift-server-todos-tutorial/blob/0f7b57bd180ff3b3ca620fdda65141712d941040/Sources/SwiftServerTodos/Entrypoint.swift#L26